tests/reals/real_tarray_undo_op.h

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef REAL_TARRAY_UNDO_OP_H
+#define REAL_TARRAY_UNDO_OP_H
+
+#include "macro_utils/macro_utils.h"
+
+#include "c_util/tarray_ll.h"
+#include "c_util/tarray.h"
+
+#include "../tarray_int/play_undo_op_types.h"
+#include "../tarray_int/play_undo_op_tarray_types.h"
+
+#define R2(X) REGISTER_GLOBAL_MOCK_HOOK(X, real_##X);
+
+#define REGISTER_TARRAY_UNDO_OP_GLOBAL_MOCK_HOOK() \
+    REGISTER_GLOBAL_MOCK_HOOK(TARRAY_ENSURE_CAPACITY(UNDO_OP), TARRAY_LL_ENSURE_CAPACITY(real_UNDO_OP)) \
```

Comment:
>UNDO_OP [](http://example.com/codeflow?start=53&length=7)

Why have this real for a "test" type here? I thought generally in the reals folder we only put re-usable reals that are needed by consumers of this lib. Seems like this is for a type defined in tarray_int.

Reply:
Your observation is awesome. 

I will move it to some other place.

It **used** to be a subfolder in tarray_int but then CMake would wine about building twice the same lib (and that does happen, as the same folder is parsed for .dll and for .exe tests). So that's why it landed in reals, but I agree this is not the right place.

Moved now.